evm/src/cpu/kernel/asm/curve_add.asm

```diff
@@ -0,0 +1,417 @@
+// #define N 0x30644e72e131a029b85045b68181585d97816a916871ca8d3c208c16d87cfd47 // BN254 base field order
+
+global ec_add:
+    // Uncomment for test inputs.
+    // PUSH 0xdeadbeef
+    // PUSH 2
+    // PUSH 1
+    // PUSH 0x1bf9384aa3f0b3ad763aee81940cacdde1af71617c06f46e11510f14f3d5d121
+    // PUSH 0xe7313274bb29566ff0c8220eb9841de1d96c2923c6a4028f7dd3c6a14cee770
+    JUMPDEST
+    // stack: x0, y0, x1, y1, retdest
+    DUP2
+    // stack: y0, x0, y0, x1, y1, retdest
+    DUP2
+    // stack: x0, y0, x0, y0, x1, y1, retdest
+    ISZERO
+    // stack: x0==0, y0, x0, y0, x1, y1, retdest
+    SWAP1
```

Comment:
nit: could avoid swapping if we checked `y0 == 0` before dup'ing `x0`

Reply:
Although on the other hand, it might be nice to add a little macro to check if the top point on the stack is the identity point, since we seem to do that 3x. If so disregard the above.